driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -616,27 +674,72 @@ public interface OidcRequestContext {
      * It does not have to be thread-safe, unless it is provided to multiple
      * MongoClients.
      */
-    public interface OidcRequestCallback {
+    public interface OidcCallback {
         /**
          * @param context The context.
          * @return The response produced by an OIDC Identity Provider
          */
-        RequestCallbackResult onRequest(OidcRequestContext context);
+        OidcCallbackResult onRequest(OidcCallbackContext context);
+    }
+
+    /**
+     * The OIDC Identity Provider's configuration that can be used to acquire an Access Token.
+     */
+    @Evolving
+    public interface IdpInfo {
+        /**
+         * @return URL which describes the Authorization Server. This identifier is the
+         * iss of provided access tokens, and is viable for RFC8414 metadata
+         * discovery and RFC9207 identification.
+         */
+        String getIssuer();
+
+        /**
+         * @return Unique client ID for this OIDC client.
+         */
+        String getClientId();
+
+        /**
+         * @return Additional scopes to request from Identity Provider. Immutable.
+         */
+        List<String> getRequestScopes();
     }
 
     /**
      * The response produced by an OIDC Identity Provider.
      */
-    public static final class RequestCallbackResult {
+    public static final class OidcCallbackResult {
 
         private final String accessToken;
 
+        @Nullable
+        private final Duration expiresIn;
+
+        @Nullable
+        private final String refreshToken;
+
+        /**
+         * @param accessToken The OIDC access token.
+         * @param expiresIn Time until the access token expires. 0 is an infinite duration.
```

Comment:
0 is not a valid `Duration` value, let's write something like the following:

```suggestion
         * @param expiresIn Time until the access token expires.
         * A {@linkplain Duration#isZero() zero-length} duration means that the access token does not expire.
```